### PR TITLE
expose pdb file from cc_common.link

### PR DESCRIPTION
**Pull request by maximyurchuk**
#9165 expose pdb file from cc_common.link

cc_common.link now has PDB file in linking_outputs.pdb_file

### CODE REVIEW

File: src/main/java/com/google/devtools/build/lib/rules/cpp/Link.java [hunk 1]
```diff
@@ -156,6 +156,14 @@ private Link() {} // uninstantiable
         ArtifactCategory.INTERFACE_LIBRARY,
         Executable.NOT_EXECUTABLE),
 
+    /** A pdb file for dynamic library. */
+    PDB_FILE(
+        LinkerOrArchiver.LINKER,
+        CppActionNames.CPP_LINK_DYNAMIC_LIBRARY,
```
Comment (maximyurchuk) [inline src/main/java/com/google/devtools/build/lib/rules/cpp/Link.java]: Don't know how to declare output for both CPP_LINK_EXECUTABLE and CPP_LINK_DYNAMIC_LIBRARY in one enum. But seems it works fine.

Can someone take an advice about right way to fix that?

Comment (maximyurchuk) [reply src/main/java/com/google/devtools/build/lib/rules/cpp/Link.java]: Actully not used now.

**Comment by maximyurchuk** [comment]
@oquenchil reasonable argumentation for me. But how close is this "future"?:) 

Anyway, I have another proposal for pdb generation in cc_common.link:
- I will manually add /PDB:foo.pdb and other flags in `cc_common.link` in `user_link_flags` argument in my code
- In that case there another field need to be added in cc_common.link for additional outputs. Because pdb output need be accessable from starlark. 

Is it good for you? If yes, I will create an issue.

**Comment by oquenchil** [comment]
A good way to solve this is a getter in linking outputs for any file/s related to debugging, whatever that is. If you want to submit a PR where linking outputs has a generic getter for these debugging files (.dwo, .pdb, etc.), with tests for Windows, Linux and Mac, then you are welcome to do it.

It will be a while before we can work on this, not in the near future.

**Comment by maximyurchuk** [comment]
@oquenchil could you explain what do you mean by "getter in linking outputs for any file/s related to debugging, whatever that is."? As I understood you said it about starlark build api, if it is true could you give an example of interface which is acceptable? For example I can change `pdb_file` in api to `symbol_files` (or `debug_files`) which can be different on different platforms (on windows it will be list with one pdb, on gcc it will be dwo file, on macos something else). If this acceptable, imho it is not necessary to implement all of these types in one pull request, because missing files on other platform can be added later.

**Comment by oquenchil** [comment]
Yes, that's exactly what I mean. A more generic name that can be used for other platforms. If we expose this, we'd like it to be exposed for debugging artifacts of every platform for which we have debugging artifacts internally. Otherwise, when Linux users see that in the API they will wonder why they are not getting any debugging files for Linux.

**Comment by maximyurchuk** [comment]
@oquenchil as I can see internally bazel has support (regarding С++) only for 
1) pdb for windows 
2) dwo/dwp for gcc


Also 'dsym' debug symbols exists on macos, but I can't find any mentions about that in rules/cpp, only in objc  rules. 

Is it acceptable to implement new API only for .dwo/.dwp and .pdb file types? (since only these types are internally suported for C++ rules for now). Or I missed some types?

**Comment by oquenchil** [comment]
Nope, that sounds good!

So a field in linker_outputs that exposes .pdb and .dwo/.dwp files. As opposed to passing the file as an input to the API and using that file.

**Comment by aiuto** [comment]
friendly ping to everyone

**Comment by maximyurchuk** [comment]
Sorry, no time for this. I suppose it is better close PR and recreate another if I return to this issue.

**Pull request closed without merging**